echidna/limit/limit_setting.py

```diff
@@ -0,0 +1,202 @@
+import numpy
+
+
+class LimitSetting(object):
+    """ Class to handle the setting of the limit.
+    """
+    def __init__(self, signal, backgrounds, data=None, **kwargs):
+        """
+        Args:
+          signal (:class: `spectra.Spectra`): signal spectrum
+          backgrounds (list): one :class: `spectra.Spectra` for each
+            background
+          data (:class: `numpy.array`): 1D data energy spectrum
+          **kwargs (dict): keyword arguments
+
+        .. note::
+
+        Keyword arguments include:
+
+          * roi (*tuple*): (energy_lower, energy_upper)
+
+        Attributes:
+          _signal (:class: `spectra.Spectra`): signal spectrum
+          _backgrounds (list): one :class: `spectra.Spectra` for each
+            background
+          _signal_config (:class: `limit_config.LimitConfig`): signal
+            configuration
+          _background_config (:class: `limit_config.LimitConfig`): 
+            background configuration
+          _calculator (:class: `chi_squared.ChiSquared`): chi squared
+            calculator to use for limit setting
+          _observed (:class: `numpy.array`): energy spectrum of observed
+            events (data)
+        """
+        self._signal = signal
+        self._signal_config = None
+        self._backgrounds = backgrounds
+        self._background_configs = {}
+        self._data = data
+        self._calculator = None
+        self._observed = None
+        if kwargs.get("roi") is not None:
+            self._roi = kwargs.get("roi")
+        else:
+            self._roi = None
+
+        # Check spectra are compatible
+        for background in backgrounds:
+            assert (background._energy_width == signal._energy_width), \
```

Comment:
Good question, I think it is typical to put exceptions in a separate file so as they are easier to locate and understand.